### PR TITLE
Cuda variant

### LINKED ISSUE #7: It looks like TensorFlow Addons is building CPU operations only
Hello Open-CE community,

It appears that TensorFlow Addons is building only the operations for CPU. From the [build documentation ](https://github.com/tensorflow/addons), there seems to be an environment variable `TF_NEED_CUDA` that needs to be set. 

There are additional flags (see below) that also need to be set about CUDA but I am guessing these might be set at high level within open-ce once the build command is triggered with CUDA.

```
export TF_NEED_CUDA="1"

# Set these if the below defaults are different on your system
export TF_CUDA_VERSION="11"
export TF_CUDNN_VERSION="8"
export CUDA_TOOLKIT_PATH="/usr/local/cuda"
export CUDNN_INSTALL_PATH="/usr/lib/x86_64-linux-gnu"
```

Would it be possible to update this recipe to build on GPU as well? Thank you very much!

Kind regards,

Pedro

### PR DESCRIPTION
**Pull request by npanpaliya**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any [tests](https://github.com/open-ce/open-ce/blob/main/tests/) to validate this change?  

## Description

Fixes #7 

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: recipe/0001-Fix-build-on-ppc.patch [hunk 1]
```diff
@@ -1,33 +1,49 @@
...
++
+ def get_tf_header_dir():
+     import tensorflow as tf
  
-@@ -118,7 +122,7 @@ def create_build_configuration():
+@@ -118,7 +138,7 @@ def create_build_configuration():
          write("build:windows --host_copt=/experimental:preprocessor")
          write("build:windows --copt=/arch=AVX")
  
 -    if is_macos() or is_linux():
```
Comment (jayfurmanek) [inline recipe/0001-Fix-build-on-ppc.patch:45]: avx is x86 specific so this should probably be just
```
 if is_linux_x86_64():
```

Comment (jayfurmanek) [reply recipe/0001-Fix-build-on-ppc.patch:45]: Also, we dont support `macos`

Comment (npanpaliya) [reply recipe/0001-Fix-build-on-ppc.patch:45]: I've not added check for macos, it was already there.

**Comment by npanpaliya** [comment]
@jayfurmanek - This is ready for review again!

**Comment by jayfurmanek** [comment]
@npanpaliya Looking back at this one.
Let's switch this PR to use the newer style meta/variant set up that is used in the Arrow recipe for adding the cuda variant there.

See https://github.com/open-ce/pyarrow-feedstock/pull/4 for details.

The benefits of this style:
 - everything is in one recipe (and one meta.yaml) making maintenance a bit easier
 - the package names remain as expected. There is no `-cpu` or `-gpu` needed in the meta package naming. The variant string `cuda` or `cpu` get put in the string name instead.
 - The main package depends on the meta/selector package, so its kind of the reverse. This is simpler:
     - `conda install tensorflow-addons` installs whatever is the `default`
     - `conda install tensorflow-addons "tensorflow-addons-proc * cuda"` installs the cuda variant
     - `conda install tensorflow-addons "tensorflow-addons-proc * cpu"` installs the cpu variant

**Comment by jayfurmanek** [comment]
replaced by #11

**Pull request closed without merging**